http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ApacheHttpClient.java

```diff
@@ -675,10 +693,10 @@ public void checkServerTrusted(X509Certificate[] x509Certificates, String s) thr
             };
         }
 
-        private SocketConfig buildSocketConfig(AttributeMap standardOptions) {
+        private SocketConfig buildSocketConfig(AttributeMap standardOptions,
+                                                  Boolean soKeepAlive) {
```

Comment:
Can we get soKeepAlive from `standardOptions.get(SdkHttpConfigurationOption.TCP_KEEPALIVE)` directly instead of passing it?